src/destination.ts

```diff
@@ -0,0 +1,91 @@
+import * as ecr from 'aws-cdk-lib/aws-ecr';
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+/**
+ * Destination information
+ */
+export interface DestinationConfig {
+  /**
+   * The URI of the destination repository to deploy to.
+   */
+  readonly destinationUri: string;
+
+  /**
+   * The tag of the deployed image.
+   * Defaults to the tag of the source if not provided.
+   */
+  readonly destinationTag?: string;
+}
+
+/**
+ * Properties needed for Source.ecr
+ */
+export interface EcrOptions {
+  /**
+   * Tag of deployed image
+   * Defaults to tag of source
+   */
+  readonly tag?: string;
+}
+
+/**
+ * Specifies docker image deployment destination
+ *
+ * Usage:
+ *
+ *  ```ts
+ *  Destination.ecr(repository, 'tag')
+ *  ```
+ *
+ */
+export abstract class Destination {
+  /**
+   * Uses an ECR repository as the destination for the image
+   *
+   * @param repository
+   * @param options
+   */
+  public static ecr(repository: ecr.IRepository, options?: EcrOptions): Destination {
+    return new EcrDestination(repository, options);
+  }
+
+  public abstract readonly config: DestinationConfig;
+
+  protected validateTag(tag: string): void {
```

Comment:
Its a docker requirement, not a registry check

Reply:
If this is always necessary, it can go into `ImageDeployment`. No need to have this functionality in the base class of all `Destination`s and require all of them to call the inherited method, if `ImageDeployment` could do the same

(It's just an opportunity for future implementors of `Destination` to forget to call this method and get it wrong)

Reply:
In other words: `Destination` should only care about things that might be *different* per destination, not about all the things that are *the same* for all destinations.